src/Quaternion.jl

```diff
@@ -381,3 +383,23 @@ function slerp(qa::Quaternion{T}, qb::Quaternion{T}, t::T) where {T}
         qa.v3 * ratio_a + qm.v3 * ratio_b,
     )
 end
+
+function sylvester(a::Quaternion, b::Quaternion, c::Quaternion)
```

Comment:
It seems the following result is not correct with type promotion.
```julia
julia> sylvester(one(Quaternion),one(Quaternion),one(Quaternion{Rational{Int}}))
QuaternionF64(-0.5, 0.0, 0.0, 0.0, false)
```
This can be solved with the following.
```suggestion
sylvester(a::Quaternion, b::Quaternion, c::Quaternion) = sylvester(promote(a,b,c)...)
function sylvester(a::Quaternion{T}, b::Quaternion{T}, c::Quaternion{T}) where {T<:Real}
```